File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/OpenOperatorHandler.scala
package edu.uci.ics.amber.engine.architecture.worker.promisehandlers

import edu.uci.ics.amber.engine.architecture.worker.DataProcessorRPCHandlerInitializer
import edu.uci.ics.amber.engine.architecture.worker.promisehandlers.OpenOperatorHandler.OpenOperator
import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand

object OpenOperatorHandler {

  final case class OpenOperator() extends ControlCommand[Unit]
}

trait OpenOperatorHandler {
  this: DataProcessorRPCHandlerInitializer =>
  registerHandler { (openOperator: OpenOperator, sender) =>
    dp.operator.open()
  }
}

File: core/amber/src/main/python/core/architecture/handlers/open_operator_handler.py
from proto.edu.uci.ics.amber.engine.architecture.worker import OpenOperatorV2
from .handler_base import Handler
from ..managers.context import Context


class OpenOperatorHandler(Handler):
    cmd = OpenOperatorV2

    def __call__(self, context: Context, command: cmd, *args, **kwargs):
        context.operator_manager.operator.open()
        return None
